test/modules/EdgeHubRestartTester/src/Settings.cs

```diff
@@ -0,0 +1,150 @@
+
+            if (!(this.DirectMethodEnable || this.MessageEnable))
+            {
+                throw new NotSupportedException("EdgeHubRestartTester requires at least one of the sending methods {DirectMethodEnable, MessageEnable} to be enable to perform the EdgeHub restarting test.");
+            }
+
+            if (restartPeriod < sdkRetryTimeout)
+            {
+                throw new InvalidDataException("sdkRetryTimeout period must be less than restartInterval period {sdkRetryTimeoutMilisec < restartPeriod}.");
+            }
+
+            if (this.RestartPeriod.Ticks < TimeSpan.FromMinutes(1).Ticks)
+            {
+                throw new InvalidDataException("RestartPeriod period must be at least one minute");
+            }
+        }
+
+        static Settings Create()
+        {
+            IConfiguration configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("config/settings.json", optional: true)
+                .AddEnvironmentVariables()
+                .Build();
+
+            return new Settings(
+                configuration.GetValue("sdkRetryTimeoutMilisec", TimeSpan.FromMilliseconds(20)),
+                configuration.GetValue<string>("IOT_HUB_CONNECTION_STRING", string.Empty),
+                configuration.GetValue<string>("IOTEDGE_DEVICEID", string.Empty),
+                configuration.GetValue<string>("reportingEndpointUrl"),
+                configuration.GetValue("restartPeriod", TimeSpan.FromMinutes(5)),
+                configuration.GetValue("testStartDelay", TimeSpan.FromMinutes(2)),
+                configuration.GetValue("testDuration", TimeSpan.Zero),
+                configuration.GetValue<bool>("directMethodEnable", false),
+                configuration.GetValue<string>("directMethodName", "HelloWorldMethod"),
+                configuration.GetValue<string>("directMethodTargetModuleId", "DirectMethodReceiver"),
+                configuration.GetValue<bool>("messageEnable", false),
+                configuration.GetValue("messageOutputEndpoint", "output1"),
+                configuration.GetValue("transportType", TransportType.Amqp_Tcp_Only),
+                configuration.GetValue<string>("IOTEDGE_MODULEID"),
+                configuration.GetValue("trackingId", string.Empty));
+        }
+
+        public string ServiceClientConnectionString { get; }
+
+        public string DeviceId { get; }
+
+        public bool DirectMethodEnable { get; }
+
+        public string DirectMethodName { get; }
+
+        public string DirectMethodTargetModuleId { get; }
+
+        public bool MessageEnable { get; }
+
+        public string MessageOutputEndpoint { get; }
+
+        public TransportType TransportType { get; }
+
+        public string ModuleId { get; }
+
+        public Uri ReportingEndpointUrl { get; }
+
+        public TimeSpan RestartPeriod { get; }
+
+        public uint SdkRetryTimeout { get; }
```

Comment:
TimeSpan

Reply:
It is a timespan from config file. But here (internal code implementation), l'd much rather have it the same unit as the SDK API with a unit in miliseconds.